src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -1049,7 +1056,7 @@ object HashMap extends ImmutableMapFactory[HashMap] with BitOperations.Int {
 
     /** The root node of the partially build hashmap */
     private var rootNode: HashMap[A, B] = HashMap.empty
-    private def plusPlusMerger = liftMerger[A, B](null).invert
+    private def plusPlusMerger = concatMerger.asInstanceOf[Merger[A, B]]
```

Comment:
maybe we should move all of the casts here
```
@inline private def plusPlusMerger[A,B] = concatMerger.asInstanceOf[Merger[A, B]]
@inline private def plusPlusMergerInverted[A,B] = concatMerger.invert.asInstanceOf[Merger[A, B]]
```
easy then to also maintain the inverted as a valif it doesnt all get inlined

This could be a followup as it doesn't affect the functionality